### ISSUE TITLE
Internal links in Jupyter Notebook doesn't work with capital letters, spaces and underscores

### ISSUE THREAD

**Issue by AndrewLaganaro** [microsoft/vscode#146352]
### Applies To

- [X] Notebooks (.ipynb files)
- [ ] Interactive Window and\/or Cell Scripts (.py files with \#%% markers)

### What happened?

Hello. As the title says, I can't make a internal link in a jupyter notebook in vscode if the cell's title starts with a capital letter, contains an underscore or contains a space, as you can see on the images.

Also, on the sole option that it works, non-capital letter header, when clicked it brings me to the bottom of the notebook, and the focus **is not** brought on the right header itself.

**Initial tests:**
![image](https://user-images.githubusercontent.com/45211616/160913693-010155ff-789e-4a5b-8956-c1547145e89c.png)

**The test headers:**
![image](https://user-images.githubusercontent.com/45211616/160914087-727a1f21-bf8b-4b30-b6c4-c3295b1bc7ef.png)

### VS Code Version

1.64.5

### Jupyter Extension Version

v2022.2.1030672458

### Jupyter logs

```shell
There are none.
```

### Coding Language and Runtime Version

Python v3.8.5

### Language Extension Version (if applicable)

Python Extension: v2022.2.1924087327

### Running Jupyter locally or remotely?

Local

**Comment by rchiodo** [comment]
In the demo notebook, the href jumps all work

![image](https://user-images.githubusercontent.com/19672699/160919104-17cdf9ef-e084-4525-bdd7-d99edd91340e.png)

But none of the header jumps do unless you rewrite them to be only lowercase.

All of the links work fine when running in 'jupyter'.

**Comment by AndrewLaganaro** [comment]
Yeah...on Vscode this notebook still with all the bugs.

Even on the links that work, they jump me to the very bottom of the notebook., not to cell 2, cell 3 or cell 4.

Also Section jump and  Section with space jump still doesn't work at all.

**Comment by mjbvz** [comment]
Does this reproduce in the [latest VS Code insiders build](https://code.visualstudio.com/insiders/)?

**Comment by rchiodo** [comment]
It did for me.

**Comment by mjbvz** [comment]
This issue was fixed in 1.68 (the May 2022 milestone)

**Comment by AndrewLaganaro** [comment]
> This issue was fixed in 1.68 (the May 2022 milestone)

got a look at https://code.visualstudio.com/updates/v1_68 and pull request 148827 ( https://github.com/microsoft/vscode/pull/148827 ) doesn't appear there.

Will it be added? Thanks

**Comment by mjbvz** [comment]
Did you test it?

**Comment by AndrewLaganaro** [comment]
> Did you test it?

Yes! everything was fixed. Capital letters, underscore and space headers now work like a charm. 

I was just asking if this specific commit ( https://github.com/microsoft/vscode/pull/148827 ) will be listed on the update website. I don't know how it works, if minor fixes won't get listed or something like that. I just wanted to show to some friends that somehow I've contributed to Vscode with my bug report 😁

Anyway, everything buggy was fixed. This issue may be closed anytime.
